src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.NumericTC.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal static partial class ValueSetFactory
+    {
+        /// <summary>
+        /// A type class providing the primitive operations needed to support a value set.
+        /// </summary>
+        /// <typeparam name="T">the underlying primitive numeric type</typeparam>
+        private interface NumericTC<T> : EqualableValueTC<T>
```

Comment:
>NumericTC [](start = 26, length = 9)

`INumericTC`?

Reply:
The convention for a typeclass is to suffix it by `TC` rather than prefix it with `I`.

Reply:
The convention for type classes is that they have a `TC` suffix and not an `I` prefix.  That will make more sense if/when we have more direct language support.  These are not ordinary interfaces.  They must be implemented only by stateless value types.

---
In reply to: [381025753](https://github.com/dotnet/roslyn/pull/41503#discussion_r381025753) [](ancestors = 381025753)